lib/ModelRoot.js

```diff
@@ -28,11 +28,14 @@ function ModelRoot(o) {
 ModelRoot.prototype.errorSelector = function errorSelector(x, y) {
     return y;
 };
-ModelRoot.prototype.comparator = function comparator(a, b) {
-    if (hasOwn(a, "value") && hasOwn(b, "value")) {
-        return a.value === b.value;
+ModelRoot.prototype.comparator = function comparator(cacheNode, messageNode) {
+    if (hasOwn(cacheNode, "value") && hasOwn(messageNode, "value")) {
```

Comment:
Well its not value equality.  Its _distinct_ checking.  Are these two items distinct.  $type-values, at least after thinking about it, are default distinct on 3 items, $type, $expires, and value.  Think of the following situation.

```javascript
model.setValue(['path'], atom(5, {$expires: Date.now() + 1000}).subscribe();
model.setValue(['path'], atom(5, {$expires: Date.now() + 2000}).subscribe();
```

Without $expires checking the second set would never happen.